File: 12. Integer to Roman/12. Integer to Roman.py
class Solution:
    def intToRoman(self, num: int) -> str:
        val : List[int] = [
            1000, 900, 500, 400,
            100,  90,  50,  40,
            10,   9,   5,   4,
            1
        ]
        syms : List[str] = [
            "M", "CM", "D", "CD",
            "C", "XC", "L", "XL",
            "X", "IX", "V", "IV",
            "I"
        ]
        result : str = ""
        for i in range(len(val)):
            while num >= val[i]:
                result += syms[i]
                num -= val[i]
        return result

File: 12. Integer to Roman/12. Integer to Roman.js
/**
 * @param {number} num
 * @return {string}
 */
var intToRoman = function(num) {
    const val = [
        1000, 900, 500, 400,
        100, 90, 50, 40,
        10, 9, 5, 4,
        1
    ]
    const syms = [
        "M", "CM", "D", "CD",
        "C", "XC", "L", "XL", 
        "X", "IX", "V", "IV", 
        "I"
    ]

    let result = ""
    const n = val.length
    for (let i = 0; i < n; i++ ){
        while (num >= val[i]){
            result = result + syms[i]
            num = num - val[i]
        }
    }

    return result
};